win32ss/gdi/ntgdi/freetype.c

```diff
@@ -4213,49 +4214,48 @@ ftGdiGetRealGlyph(
     INT error;
     FT_GlyphSlot glyph;
     FT_BitmapGlyph realglyph;
-    DWORD cdw;
 
     ASSERT_FREETYPE_LOCK_HELD();
 
-    if (Cache->Aspect.EmuBoldItalic)
+    if (Cache->Hashed.Aspect.EmuBoldItalic)
     {
-        error = FT_Load_Glyph(Cache->Face, Cache->GlyphIndex, FT_LOAD_NO_BITMAP);
+        error = FT_Load_Glyph(Cache->Hashed.Face, Cache->Hashed.GlyphIndex, FT_LOAD_NO_BITMAP);
         if (error)
         {
-            DPRINT1("WARNING: Failed to load and render glyph! [index: %d]\n", Cache->GlyphIndex);
+            DPRINT1("WARNING: Failed to load and render glyph! [index: %d]\n",
+                    Cache->Hashed.GlyphIndex);
             return NULL;
         }
 
-        glyph = Cache->Face->glyph;
+        glyph = Cache->Hashed.Face->glyph;
 
-        if (Cache->Aspect.Emu.Bold)
+        if (Cache->Hashed.Aspect.Emu.Bold)
             FT_GlyphSlot_Embolden(glyph);
-        if (Cache->Aspect.Emu.Italic)
+        if (Cache->Hashed.Aspect.Emu.Italic)
             FT_GlyphSlot_Oblique(glyph);
-        realglyph = ftGdiGlyphSet(Cache->Face, glyph, Cache->Aspect.RenderMode);
+        realglyph = ftGdiGlyphSet(Cache->Hashed.Face, glyph, Cache->Hashed.Aspect.RenderMode);
     }
     else
     {
-        cdw = (sizeof(FONT_CACHE_ENTRY) - offsetof(FONT_CACHE_ENTRY, GlyphIndex)) / sizeof(DWORD);
-        Cache->dwHash = IntGetHash(&Cache->GlyphIndex, cdw);
+        Cache->dwHash = IntGetHash(&Cache->Hashed, sizeof(Cache->Hashed) / sizeof(DWORD));
```

Comment:
Code around line 3261 could also be improved similarly.

Reply:
```c
    NewEntry->BitmapGlyph = BitmapGlyph;
    NewEntry->dwHash = Cache->dwHash;
    NewEntry->Hashed = Cache->Hashed;
```